spec/views/topics/show.html.erb_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "ostruct"
+
+describe "topics/show.html.erb" do
+  fab!(:topic) { Fabricate(:topic, subtype: Topic::QA_SUBTYPE) }
+  fab!(:post) { Fabricate(:post, topic: topic) }
+  fab!(:qa_comment) { Fabricate(:qa_comment, raw: "this is a comment!", post: post) }
+
+  it "includes QA comments for crawler view" do
+    topic_view = OpenStruct.new(
+      topic: topic,
+      posts: [post],
+      comments: [qa_comment]
+    )
+
+    topic_view.stubs(:summary).returns('')
+    topic_view.stubs(:link_counts).returns([])
+    topic_view.stubs(:post_custom_fields).returns('')
+    topic_view.stubs(:filtered_posts).returns(["post_1", "post_2"])
```

Comment:
Can you try using a request spec instead which I think will allow you to avoid all the stubbing in the test.

Reply:
Good idea! Done.